orttraining/orttraining/training_api/module.cc

```diff
@@ -0,0 +1,252 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include "core/framework/execution_provider.h"
+#include "core/session/inference_session.h"
+#include "core/session/environment.h"
+
+#include "orttraining/training_api/include/module.h"
+#include "orttraining/training_api/include/utils.h"
+
+using namespace onnxruntime;
+
+namespace onnxruntime {
+namespace training {
+namespace api {
+
+namespace {
+
+// TODO: consolidate with frontend tooling
```

Comment:
Yes, this TODO is to avoid having multiple sources of truth for the input name. It implies we should find a way to have a single function that can get the name for both offling tooling and online training step.

Reply:
The other thing to note is that the `Module::ResetGrad` does only lazy reset, so we might want to expose a way that does explicit reset. This might be a good place to do that. I think we should keep the `ResetGrad` here and document what it does (and also add support for GPU).